manifests/repo.pp

```diff
@@ -2,30 +2,39 @@
 #
 # @param manage_container_manager whether to add cri-o repository or not
 # @param crio_version version o cri-o
+# @param container_manager The name of the container manager
 #
 class k8s::repo (
-  Boolean $manage_container_manager = $k8s::manage_container_manager,
-  String[1] $crio_version           = $k8s::version.split('\.')[0, 2].join('.'),
+  Boolean $manage_container_manager          = $k8s::manage_container_manager,
+  K8s::Container_runtimes $container_manager = $k8s::container_manager,
+  String[1] $crio_version                    = $k8s::version.split('\.')[0, 2].join('.'),
 ) {
   case fact('os.family') {
     'Debian': {
-      if fact('os.name') == 'Debian' {
-        if Integer(fact('os.release.major')) < 10 {
-          warning('CRI-O is only available for Debian 10 and newer')
-          warning('containerD is only available for Debian 10 and newer')
+      case fact('os.name') {
+        'Debian': {
+          if Integer(fact('os.release.major')) < 10 {
```

Comment:
metadata.json  doesn't list Debian 9, so we could remove this if block.

Reply:
@ananace what do you think, can we remove this, or should we keep it? i for myself don't do debian.

Reply:
I think that's reasonable, Debian 9 is quite old after all

Reply:
okay, than i will remove this part